backend/tracim_backend/lib/collaborative_document_edition/collaboration_document_edition.py

```diff
@@ -0,0 +1,124 @@
+import mimetypes
+import os
+from os.path import isfile
+from os.path import join
+import typing
+
+from defusedxml import ElementTree
+import requests
+from sqlalchemy.orm import Session
+
+from tracim_backend import CFG
+from tracim_backend.exceptions import FileTemplateNotAvailable
+from tracim_backend.exceptions import NotAFileError
+from tracim_backend.exceptions import NotReadableFile
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionConfig,
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionFileType,
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionToken,
+)
+from tracim_backend.lib.collaborative_document_edition.models import FileTemplateList
+from tracim_backend.lib.core.content import ContentApi
+from tracim_backend.lib.utils.utils import is_dir_exist
+from tracim_backend.lib.utils.utils import is_dir_readable
+from tracim_backend.lib.utils.utils import is_file_exist
+from tracim_backend.lib.utils.utils import is_file_readable
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import Content
+
+
+class CollaborativeDocumentEditionApi(object):
+    def __init__(self, current_user: typing.Optional[User], session: Session, config: CFG) -> None:
+        self._session = session
+        self._user = current_user
+        self._config = config
+
+    def discover(self) -> typing.List[CollaborativeDocumentEditionFileType]:
+        response = requests.get(
+            self._config.COLLABORATIVE_DOCUMENT_EDITION__COLLABORA__BASE_URL + "/hosting/discovery",
+            timeout=2,
+        )
+        root = ElementTree.fromstring(response.text)
+        supported_collabora_file = []  # type: typing.List[CollaborativeDocumentEditionFileType]
+        for xml_app in root.findall("net-zone/app"):
+            mimetype = xml_app.get("name")
+            # INFO - G.M - 2019-07-17 - this list return also a non mimetype type,
+            # we should not add him.
+            if mimetype == "Capabilities":
+                continue
+            for xml_action in xml_app:
+                extension = xml_action.get("ext")
+                url_source = xml_action.get("urlsrc")
+                associated_action = xml_action.get("name")
+                supported_collabora_file.append(
+                    CollaborativeDocumentEditionFileType(
+                        mimetype=mimetype,
+                        extension=extension,
+                        associated_action=associated_action,
+                        url_source=url_source,
+                    )
+                )
+        return supported_collabora_file
+
+    def get_collaboration_document_edition_config(self) -> CollaborativeDocumentEditionConfig:
```

Comment:
We are in a class `CollaborativeDocumentEditionApi`, so the naming `get_collaboration_document_edition_config` would better be  `get_config()`, no?

Reply:
done